File: features/rolemanagement/src/main/java/com/nivasafinance/features/rolemanagement/role/repository/RoleRepository.java
package com.nivasafinance.features.rolemanagement.role.repository;

import com.nivasafinance.features.rolemanagement.role.entity.Role;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface RoleRepository extends JpaRepository<Role, Long> {
    List<Role> findByNameIn(List<String> names);
}

File: features/rolemanagement/src/main/java/com/nivasafinance/features/rolemanagement/role/repository/RoleRepository.kt
package com.nivasafinance.features.rolemanagement.role.repository

import com.nivasafinance.features.rolemanagement.role.entity.Role
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface RoleRepository : JpaRepository<Role, UUID> {
    fun findByNameIn(names: List<String>): List<Role>
}
